mlflow/pipelines/steps/predict.py

```diff
@@ -107,6 +107,27 @@ def _run(self, output_directory):
                 error_code=BAD_REQUEST,
             )
 
+        # check if output location is already populated as we allow no overwrites
+        output_format = self.step_config["output_format"]
+        output_location = self.step_config["output_location"]
+        if output_format == "parquet" or output_format == "delta":
+            output_populated = os.path.exists(output_location)
+        else:
+            try:
+                output_populated = spark._jsparkSession.catalog().tableExists(output_location)
```

Comment:
There is a `spark.catalog.tableExists` method, but this method is only available in pyspark 3.3.0. Using this method instead, which is available in spark 2.1.x or later.